spring-integration-core/src/main/java/org/springframework/integration/handler/DelayHandler.java

```diff
@@ -205,6 +226,71 @@ public void setDelayedAdviceChain(List<Advice> delayedAdviceChain) {
 		this.delayedAdviceChain = delayedAdviceChain;
 	}
 
+	/**
+	 * Set a message channel to which an {@link ErrorMessage} will be sent if sending the
+	 * released message fails. If the error flow returns normally, the release is
+	 * complete. If the error flow throws an exception, the release will be re-attempted.
+	 * If there is a transaction advice on the release task, the error flow is called
+	 * within the transaction.
+	 * @param delayedMessageErrorChannel the channel.
+	 * @see #setMaxAttempts(int)
+	 * @see #setRetryDelay(long)
+	 * @since 5.0.8
+	 */
+	public void setDelayedMessageErrorChannel(MessageChannel delayedMessageErrorChannel) {
```

Comment:
Can't we come up with some `RetryInterceptor` which could reuse our `ErrorMessageSendingRecoverer` instead of all these retry configuration details?

This one could be added to the `delayedAdvice()` alongside with the `TransactionalInterceptor`. The same way how we do a `request-handler-advice-chain`

Thanks

Reply:
No; we can't do that because the retry advice (even with a stateful interceptor) relies on blocking the thread, which we don't want to do; scheduler threads are precious. This mechanism implements delayed retries using the scheduler and doesn't block any threads.

Reply:
OK. Sounds interesting. 
I need to digest it a bit to get used to...

Can we still try to reuse our `ErrorMessagePublisher`  at least?

Reply:
I started looking at it, but it adds quite a bit of complexity.

We should probably revisit it (and the `ErrorMessageStrategy`) to make them easier to reuse for cases like this, but I don't really want to do that as part of this PR.